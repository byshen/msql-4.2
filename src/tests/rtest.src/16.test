#
# Test the millitime type
#

create table millitime(v1 millitime)\p\g
create index idx1 on millitime(v1)\p\g

INSERT INTO millitime  VALUES ('5:30:20.123')\p\g
INSERT INTO millitime  VALUES ('2:17:18.654')\p\g
INSERT INTO millitime  VALUES ('17:16:15.140')\p\g
INSERT INTO millitime  VALUES ('23:59:59.999')\p\g
INSERT INTO millitime  VALUES ('0:00:00.001')\p\g
INSERT INTO millitime  VALUES ('10:10:10')\p\g
INSERT INTO millitime  VALUES ('23:61:00.000')\p\g

select * from millitime order by v1\p\g
select * from millitime where v1 = '17:16:15.140'\p\g
select * from millitime where v1 = '10:10:10'\p\g
select * from millitime where v1 between '5:00:00.000' and '20:00:00.000'\p\g
