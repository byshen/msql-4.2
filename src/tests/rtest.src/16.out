
Welcome to the miniSQL monitor.  Type \h for help.


mSQL >     -> 
Query buffer
------------

create table millitime(v1 millitime)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
create index idx1 on millitime(v1)
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL >     -> 
Query buffer
------------
INSERT INTO millitime  VALUES ('5:30:20.123')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO millitime  VALUES ('2:17:18.654')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO millitime  VALUES ('17:16:15.140')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO millitime  VALUES ('23:59:59.999')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO millitime  VALUES ('0:00:00.001')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO millitime  VALUES ('10:10:10')
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.




mSQL > 
Query buffer
------------
INSERT INTO millitime  VALUES ('23:61:00.000')
[continue]
    -> 

ERROR : Invalid mill-time format


mSQL >     -> 
Query buffer
------------
select * from millitime order by v1
[continue]
    -> 
Query OK.  6 row(s) modified or retrieved.

 +---------------+
 | v1            |
 +---------------+
 | 00:00:00.001  |
 | 02:17:18.654  |
 | 05:30:20.123  |
 | 10:10:10.000  |
 | 17:16:15.140  |
 | 23:59:59.999  |
 +---------------+



mSQL > 
Query buffer
------------
select * from millitime where v1 = '17:16:15.140'
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +---------------+
 | v1            |
 +---------------+
 | 17:16:15.140  |
 +---------------+



mSQL > 
Query buffer
------------
select * from millitime where v1 = '10:10:10'
[continue]
    -> 
Query OK.  1 row(s) modified or retrieved.

 +---------------+
 | v1            |
 +---------------+
 | 10:10:10.000  |
 +---------------+



mSQL > 
Query buffer
------------
select * from millitime where v1 between '5:00:00.000' and '20:00:00.000'
[continue]
    -> 
Query OK.  3 row(s) modified or retrieved.

 +---------------+
 | v1            |
 +---------------+
 | 05:30:20.123  |
 | 10:10:10.000  |
 | 17:16:15.140  |
 +---------------+



mSQL > 
Bye!

